lib/plugins/aws/package/compile/events/apiGateway/lib/restApi.js

```diff
@@ -139,6 +138,77 @@ module.exports = {
       );
     }
 
+    // Models are defined in the provider and can be used in http events later
+    if (!_.isUndefined(apiGateway.schemas)) {
```

Comment:
I would also seclude it into dedicated `lib/plugins/aws/package/compile/events/apiGateway/lib/requestValidators.js` (and handle generation of all `AWS::ApiGateway::Model` and `AWS::ApiGateway::RequestValidator` resources, so it's in one place)